.travis/build-doc.sh

```diff
@@ -12,6 +12,15 @@ if ! travis_isPush; then
     exit 0
 fi
 
+
+#
+# First step: build pmd with profile "generate-rule-docs"
+# The docs should appear under "docs/pages/rules/..." for each language
+#
+./mvnw clean verify -Dmaven.test.skip=true -Dmaven.javadoc.skip=true -P generate-rule-docs
```

Comment:
TODO: since we are executing the link checker only with the profile "generate-rule-docs", we need to make sure, that we build the docs also for pull-requests.

We just don't upload the generate docs then for PRs - only for our own pushes.

Reply:
Done in 6e6adb9